### DISCUSSION TITLE
CUDA out of memory

### DISCUSSION THREAD

**Discussion by saiakarsh193** [coqui-ai/TTS#2370] [General Q&A]
When I am running the training script (ljspeech: tacotron2-DCA), I am getting this error 

```
torch.cuda.OutOfMemoryError: CUDA out of memory. Tried to allocate 14.00 MiB (GPU 0; 10.76 GiB total capacity; 
9.05 GiB already allocated; 6.69 MiB free; 9.63 GiB reserved in total by PyTorch) 
If reserved memory is >> allocated memory try setting max_split_size_mb to avoid fragmentation.  
See documentation for Memory Management and PYTORCH_CUDA_ALLOC_CONF.
```

I am running the code (trainer.distribute) on a compute cluster with 4 GPUs each having 8GB VRAM.
(This happens even when i run the default code on a single GPU!).
I never encounter such an issue when running other repos for a similar task. 

It does advise me to set max_split_size_mb. Can someone explain how and where to do that.
I would really appreciate it if anyone can explain why all GPU are getting maxed out because i am not able to understand why we would need over 8 GB, and how the above solution can help.

Thanks

**Comment by eloop001** [comment]
I'm new, but I can actually answer this one. If you look in 
/root/TTS/recipes/thorsten_DE/tacotron2-DDC/train_tacotron_ddc.py

Thorsten actually mentions this in the file.

Line 42: batch_size=40,  # BS of 40 and max length of 10s will use about 20GB of GPU memory
Line 68:     max_audio_len=22050 * 10,

So with 8 GB memory you either have to shorten your wav files, or decrease the batch size. It doesn't matter if there is two cards with 8 GB each -it will not add up to 16GB. I know because I'm currently running a training on Google Cloud with 4xT4 cards. they have 16GB each and I ran into the same problem, where I had to adjust. Just for fun, I grabbed an A100 (with 40 GB of V-RAM) running on a 30 core machine with 256 Gig of memory....aaannd ran straight into the same problem, because my sample were to big, in relation to the batch size and max_audio_length.

I recommend splitting up your dataset into small segments, of possible. 

The reason for using that much memory is, that the training sets up a matrix of size samples x batch size... or something to that effect..
